community/kernel/src/main/java/org/neo4j/kernel/recovery/LogTailScanner.java

```diff
@@ -248,26 +258,71 @@ public LogTailInformation getTailInformation() throws UnderlyingStorageException
         return logTailInformation;
     }
 
+    static class ExtractedTransactionRecord
+    {
+        private long id;
+        private boolean failure;
+
+        ExtractedTransactionRecord()
+        {
+            id = NO_TRANSACTION_ID;
+            failure = false;
+        }
+
+        public long getId()
+        {
+            return id;
+        }
+
+        public void setId( long id )
+        {
+            this.id = id;
+        }
+
+        public boolean isFailure()
+        {
+            return failure;
+        }
+
+        public void setFailure( boolean failure )
```

Comment:
I don't see why this class needs to have publically accessable setters instead of final fields and set them in constructor.